src/lib/y2users/config.rb

```diff
@@ -92,6 +92,16 @@ def clone
       config
     end
 
+    # Merges elements (users and groups) with elements from another config
+    #
+    # @note The caller config is modified.
+    #
+    # @return [Config]
+    def merge(other)
```

Comment:
maybe as convention I would call it `merge!` so it is clear that it is modified and consistent with Hash.

Another topic is that merged. I would do it as
```
    def merge(other, merger: Mergers::Replace.new)
       merger.merge(self, other)
    end
```
this way you can pass merger with different strategy.